### ISSUE TITLE
masked character sequence pretrainer

### ISSUE THREAD

**Issue by jamesmf** [jamesmf/cclm#20]
pretraining task for the base of a model.

input_string --> spatial dropout, noise data augmentation --> input_string

**Comment by jamesmf** [comment]
spatial dropout 1d isn't appropriate for dropping whole timestep-features, just whole 1D features. Pass an explicit mask during training instead.

**Comment by jamesmf** [comment]
`BasePretrainer`, while possibly requiring a rename, implements this by putting a trivial layer on top of the base and doing  character-level masking.
